tests/unit/amo/components/TestUserProfileEdit.js

```diff
@@ -142,6 +154,31 @@ describe(__filename, () => {
     expect(root.find('.UserProfileEdit-deletion-modal')).toHaveLength(0);
   });
 
+  it('renders user profile page for current logged-in user (non-developer)', () => {
```

Comment:
maybe to be a little clearer we could state what is different for developer case in description ~ `renders user profile page with limited fields for current logged-in user (non-developer)`  ?